yt/data_objects/static_output.py

```diff
@@ -861,22 +862,60 @@ def _setup_particle_types(self, ptypes=None):
     _last_finfo = None
 
     def _get_field_info(self, ftype, fname=None):
-        field_info, is_ambiguous = self._get_field_info_helper(ftype, fname)
+        field_info, candidates = self._get_field_info_helper(ftype, fname)
 
         if field_info.name[1] in ("px", "py", "pz", "pdx", "pdy", "pdz"):
-            # escape early as as bandaid solution to
+            # escape early as a bandaid solution to
             # https://github.com/yt-project/yt/issues/3381
             return field_info
 
-        if is_ambiguous:
+        def _are_ambiguous(candidates: List[Tuple[str, str]]) -> bool:
+            if len(candidates) < 2:
+                return False
+
+            ftypes, fnames = (list(_) for _ in unzip(candidates))
+            assert all(name == fnames[0] for name in fnames)
+
+            fi = self.field_info
+
+            all_aliases: bool = all(
+                fi[c].is_alias_to(fi[candidates[0]]) for c in candidates
+            )
+
+            all_equivalent_particle_fields: bool
+            if (
+                self.particle_types is None
+                or self.particle_unions is None
+                or self.particle_types_raw is None
+            ):
+                all_equivalent_particle_fields = False
+            elif all(ft in self.particle_types for ft in ftypes):
+                ptypes = ftypes
+
+                sub_types_list: List[Set[str]] = []
+                for pt in ptypes:
+                    if pt in self.particle_types_raw:
+                        sub_types_list.append({pt})
+                    elif pt in self.particle_unions:
+                        sub_types_list.append(set(self.particle_unions[pt].sub_types))
+                all_equivalent_particle_fields = all(
+                    st == sub_types_list[0] for st in sub_types_list
+                )
+            else:
+                all_equivalent_particle_fields = False
+
+            return not (all_aliases or all_equivalent_particle_fields)
+
+        if _are_ambiguous(candidates):
             ft, fn = field_info.name
+            possible_ftypes = [c[0] for c in candidates]
             msg = (
-                f"The requested field name '{fn}' "
+                f"The requested field name {fn!r} "
                 "is ambiguous and corresponds to any one of "
-                f"the following field types:\n {self.field_info._ambiguous_field_names[fn]}\n"
+                f"the following field types:\n {possible_ftypes}\n"
```

Comment:
With this code, isn't it possible that `possible_ftypes` has multiple time the same value, e.g. if `candidates = [("foo", "bar"), ("foo", "baz")]` ?